UPGRADE.md

```diff
@@ -0,0 +1,118 @@
+# Upgrade
+
+All changes required for users to upgrade to newer versions of this library will be documented or referenced here.
+
+## 2.* -> 3.0.0
+
+### `string`, `int`, `float`, `bool` are no longer ignored
+
+In 2.x, a `Zend\Code\Generator\ParameterGenerator` with name `foo` and type 
+`string`, `int`, `float` or `bool` simply generated code `"$foo"`:
+
+```php
+$generator = new \Zend\Code\ParameterGenerator('foo');
+
+$generator->setType('string');
+
+echo $generator->generate(); // "$foo"
+```
+
+In 3.x, this code will instead produce `"string $foo"`.
+If you generate code that should run in PHP 5.x, it is advisable to strip
+`string`, `int`, `float` and `bool` from type definitions passed to
+`Zend\Code\ParameterGenerator` instances.
+
+### `Zend\Code\Reflection\ParameterReflection#getType()` changes
+
+PHP 7 introduced [`ReflectionParameter#getType()`](http://php.net/manual/en/reflectionparameter.gettype.php).
+
+In order to not override this method, `Zend\Code\Reflection\ParameterReflection#getType()`
+was renamed to `Zend\Code\Reflection\ParameterReflection#detectType()`.
+
+If you relied on `Zend\Code\Reflection\ParameterReflection#getType()`, you can
+simply replace the method calls in your code.
+ 
+### DocBlock types ignored by `Zend\Code\Generator\ParameterGenerator::fromReflection()`
+
+As a direct consequence of the previous change, calls to 
+`Zend\Code\Generator\ParameterGenerator::fromReflection()` will not mirror the
+type hints read from a method's DocBlock.
+
+As an example, take following code:
+
+```php
+class Foo
+{
+    /**
+     * @param string $baz
+     */
+    public function bar($baz)
+    {
+    }
+}
+
+$methodGenerator = \Zend\Code\Generator\MethodGenerator::fromReflection(
+    new \Zend\Code\Reflection\MethodReflection('Foo', 'bar')
+);
+
+var_dump($methodGenerator->getParameters()[0]->getType());
+```
+
+In version 2.x, this code produces `"string"`, in version 3.x it returns `null`.
```

Comment:
It also looks like this functionality *is* still present:

- https://github.com/zendframework/zend-code/pull/30/files#diff-571bf3d3af7801a30d8cc91a9677dddfR101

Can you verify?